File: src/main/java/com/pocketempire/units/UnitFactory.java
package com.pocketempire.units;

import com.pocketempire.config.UnitConfigLoader;
import com.pocketempire.entities.Unit;
import com.pocketempire.units.UnitType;

public class UnitFactory {
    public static Unit create(UnitType type, String id, String name, int q, int r, String factionId) {
        UnitStats stats = UnitConfigLoader.getConfig(type.name());

        return new Unit.Builder(id, q, r, factionId)
                .name(name)
                .config(stats)
                .unitType(type)
                .build();
    }
}

File: src/main/kotlin/com/pocketempire/units/UnitFactory.kt
package com.pocketempire.units

import com.pocketempire.config.UnitConfigLoader
import com.pocketempire.entities.Unit

object UnitFactory {

    @JvmStatic
    fun create(
        type: UnitType,
        id: String,
        name: String,
        q: Int,
        r: Int,
        factionId: String
    ): Unit = createUnit(type, id, name, q, r, factionId)

    @JvmStatic
    fun createUnit(
        type: UnitType,
        id: String,
        name: String,
        q: Int,
        r: Int,
        factionId: String
    ): Unit {

        val stats = UnitConfigLoader.getConfig(type.name)

        return Unit.Builder(id, q, r, factionId)
            .name(name)
            .config(stats)
            .unitType(type)
            .build()
    }
}
